pkg/utils/utils.go

```diff
@@ -240,3 +240,40 @@ func RemoveString(slice []string, s string) (result []string) {
 	}
 	return
 }
+
+func EnvValueEqual(cenv, denv []v1.EnvVar) bool {
+	var found bool
+	if len(cenv) != len(denv) {
+		return false
+	}
+	delemcheck := make([]bool, len(denv))
+	for _, celem := range cenv {
+		found = false
+		if celem.ValueFrom != nil {
+			continue
+		}
+		for d, delem := range denv {
+			if delem.ValueFrom != nil {
+				delemcheck[d] = true
+				continue
+			}
+			if celem.Name == delem.Name {
+				delemcheck[d] = true
+				if celem.Value != delem.Value {
+					return false
+				}
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	for _, dc := range delemcheck {
```

Comment:
is this necessary since you're checking the array lengths and return `false` if `!found`

Reply:
Actually, the all delemcheck codes were _not_ necessary...  It was meant for delem was superset of celem, but it's handled by len(celem) != len(delem).  So, I'm deleting them.  Thanks!